dace/frontend/fortran/ast_transforms.py

```diff
@@ -3655,6 +3668,17 @@ def visit_Execution_Part_Node(self, node: ast_internal_classes.Execution_Part_No
             self.nodes_to_process = {}
             n = self.visit(child)
 
+            # Restart type inference to ensure that we reassign size
```

Comment:
Out of curiosity without a full understanding of the type inference logic: Is it possible to get into an infinite loop of constantly reassigning the size of the same temporaries over and over?

Reply:
I don't think so, this part is not done in a loop.